### ISSUE TITLE
Polish National Focus Tree

### ISSUE THREAD

**Issue by vapinus** [Wolferos/Hearts-of-Iron-IV-The-Great-War#260]
Hello everyone
I am very glad that finally there is a mod that can be played in the interwar period, in particular the years 1919-1920, when new countries were created in Central Europe. There will also be a chance to play as Whites or Reds in the Russian Civil War. Very interesting. 

More than 10 years ago, I collaborated on making a mod for Hoi2 regarding the Polish-Bolshevik war. Unfortunately, before the mod was ready, HoI3 was released. But I was left with a lot of materials about Poland, the Baltic States, White and Red Russia and the Don Cossacks. In addition, I am a historian, so I also have a reference library :) I have no shortage of sources.

I looked at all the files (I think?) available for the 1919 scenario. And I started working right away. I've already corrected a few things. I have already written a few of my comments to Wolferos.

But now I'm preparing a brand new focus tree for Poland, because the one in the game doesn't reflect the events that took place in Poland in 1919. And later too. It is a fascinating challenge to make a Polish focus tree for the entire interwar period.

As I wrote, I am a historian, but not a modding master. I can do an NF tree, simple triggers, events, and so on. But I can't do more complex things anymore. That's why I want to share my ideas and comments to use in building your mod. I am here to help. 

Let the force be with you :)
I am attaching a few screenshots.
Sorry for my poor english ;)

**Comment by vapinus** [comment]
screenshots soon

**Comment by vapinus** [comment]
![1 the whole tree](https://user-images.githubusercontent.com/69906831/109831799-10690000-7c40-11eb-9181-704dd5e0715f.png)

![2 political branch1](https://user-images.githubusercontent.com/69906831/109831853-1c54c200-7c40-11eb-8bea-e032b4b4c337.png)

![3 political branch2](https://user-images.githubusercontent.com/69906831/109831882-224aa300-7c40-11eb-9295-0ac384ef5f51.png)

![4 political branch and economy](https://user-images.githubusercontent.com/69906831/109831931-2e366500-7c40-11eb-9d99-1b1c849b08ea.png)

**Comment by vapinus** [comment]
![5 army_navy_air2](https://user-images.githubusercontent.com/69906831/109832051-4dcd8d80-7c40-11eb-9260-eacf0d54b15a.png)

![6 army_navy_air1](https://user-images.githubusercontent.com/69906831/109832085-54f49b80-7c40-11eb-8105-a806463c3203.png)

![7 political_economy_army](https://user-images.githubusercontent.com/69906831/109832116-5aea7c80-7c40-11eb-8786-65af8ff1d2c3.png)

![8 this is how the map of Poland should look like on July 1, 1919](https://user-images.githubusercontent.com/69906831/109832143-6047c700-7c40-11eb-815c-bf209bf379e6.png)

**Comment by vapinus** [comment]
Of course, this is a draft. There are no national focuses for IIWW there yet, especially the Polish government-in-exile.

**Comment by vapinus** [comment]
And it could be useful to improve the map. Without the state "Free City of Gdańsk", the history of the Polish-German conflict over Gdańsk will be completely unclear, as in Vanilla.

**Comment by vapinus** [comment]
I found this idea: POL_interwar_spirit_WielkopolskaUprising.

The Greater Poland Uprising began in December 1918 and ended with a ceasefire on February 16, 1919, signed in Trier. The fight was between Poles and Germans. It has nothing to do with the Bolsheviks. Where did the name "Beat the Bolshevik" come from?

https://en.wikipedia.org/wiki/Greater_Poland_uprising_(1918%E2%80%931919)

So the idea of POL_interwar_spirit_WielkopolskaUprising should not work from the moment you start the game. The uprising is over.

**Comment by vapinus** [comment]
Nobody has any comments? No objections?

I'm still working on a modification. I want to create a focus tree so that the player can choose what Poland he wants to create: large, small, national, federal, right-wing or left-wing. With Piłsudski at the head, or Dmowski, even with someone else. That's why I built the focus tree from scratch.

Now I'm working on events and decisions and the text.

**Comment by NicolasGrosjean** [comment]
The land military tree should be developed deeper, for example with

- focus to licence Entente material (like Renault T-17) like Yougoslavia in the Vanilla
- given some technology bonus to allow the player to specialize his army

I have found other mods on the interbellum : 

- [Interbellum - The 1920s](https://steamcommunity.com/sharedfiles/filedetails/?id=1561451947)
- [1920 - Aftermath of the Great War](https://discord.gg/YXP6vdvJSP)

I don't know if and how they can share their progress to avoid to do 3 very incompleted versions.

**Comment by DevilIris** [comment]
I will start coding the focus tree. Do you have any progress with code part?

**Comment by vapinus** [comment]
Here is an error in file 402-Akhtubinsk.txt - UKR controll prov 3562

state={
	id=402
	name="STATE_402"
	manpower = 282245
	state_category=rural
	provinces={ 1220 1247 1314 1371 1429 4227 4254 4294 4300 4303 4374 4376 4394 7243 7313 7337 7341 7354 7370 7410 10131 10164 10253 10264 12108 12179 12227 12235 12254 12294 }
	resources={
		wood=0
		chromium=31
		}
	history={
		owner=RUS
		add_core_of=RUS
		1936.1.1={ 
			owner=SOV
			add_core_of=SOV
			buildings={
				infrastructure=3
				}
			}
		buildings={
			infrastructure=3
			}
		1919.1.1 = {
            add_core_of = SOV

            UKR = {
				set_province_controller = 3562			
			}
        }
		}
	}

Similar error is in file 233-Caucasus Mountains.txt: POL controlls prov 3759

**Comment by vapinus** [comment]
> 
> 
> I will start coding the focus tree. Do you have any progress with code part?

What do you mean that you will **start coding** the focus tree?

I'm working on the focus tree all the time. I keep changing something. It is still a long way to the end of work.

**Comment by vapinus** [comment]
map - as of July 1, 1919, work in progress... now I check which part of Russia was ruled by the Reds and which by the Whites.
![2021 04 07-21 21](https://user-images.githubusercontent.com/69906831/113922497-a23bce00-97e7-11eb-8bea-20d939927cf4.png)

**Comment by vapinus** [comment]
Does anyone know what the "unordered" division name list should look like in the TAG_names_divisions.txt file?

**Comment by vapinus** [comment]
Can anyone tell Wolferos to look at the news on Steam?
